webmessaging/MessageEvent-trusted.html

```diff
@@ -0,0 +1,44 @@
+<!DOCTYPE html>
+<title>MessagePort message events are trusted</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<link rel="help" href="https://html.spec.whatwg.org/multipage/comms.html#dom-messageport-postmessage">
+<!-- See also:
+- https://github.com/whatwg/html/issues/1602
+- https://github.com/whatwg/html/pull/1935
+-->
+
+<script>
+"use strict";
+
+async_test(t => {
+  assert_true("MessageChannel" in self, "The browser must support MessageChannel");
+
+  const channel = new MessageChannel();
+
+  channel.port2.onmessage = t.step_func_done(e => {
+    assert_equals(e.isTrusted, true);
+  });
+
+  channel.port1.postMessage("ping");
+}, "With a MessageChannel and its MessagePorts");
+
+async_test(t => {
+  assert_true("BroadcastChannel" in self, "The browser must support BroadcastChannel");
+
+  new Worker("MessageEvent-trusted-worker.js");
```

Comment:
It is probably safer to not create this worker until after you've created the BroadcastChannel instance and added the onmessage event handler. I don't think there is anything that guarantees the worker won't be created, ran, and posts its message to the channel before this method has a chance to connect to the channel itself (unless I'm missing something in how cross-thread-synchronization is specified in the html spec)